# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/ipc/win/handler/AccessibleHandlerControl.h
# Commit: f262a231f046
# Full Hash: f262a231f0460c4295999b158a6b00868b49b843
# Author: James Teh <jteh@mozilla.com>
# Date: 2020-06-01 16:21:08
# Description:
#   Bug 1642141: When clearing the AccessibleHandlerControl Accessible cache, swap it into a temporary map first to avoid re-entry into the main map while clearing. r=MarcoZ
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D77623
# ==============================================================================

diff -r 2692b8d159d5 -r f262a231f046 accessible/ipc/win/handler/AccessibleHandlerControl.h
--- a/accessible/ipc/win/handler/AccessibleHandlerControl.h	Mon Jun 01 05:13:33 2020 +0000
+++ b/accessible/ipc/win/handler/AccessibleHandlerControl.h	Mon Jun 01 05:45:44 2020 +0000
@@ -85,7 +85,8 @@
   UniquePtr<mscom::RegisteredProxy> mIA2Proxy;
   UniquePtr<mscom::RegisteredProxy> mHandlerProxy;
   // We can't use Gecko APIs in this dll, hence the use of std::unordered_map.
-  std::unordered_map<long, RefPtr<AccessibleHandler>> mAccessibleCache;
+  typedef std::unordered_map<long, RefPtr<AccessibleHandler>> AccessibleCache;
+  AccessibleCache mAccessibleCache;
 };
 
 extern mscom::SingletonFactory<AccessibleHandlerControl> gControlFactory;
